social_core/backends/discourse.py

```diff
@@ -0,0 +1,86 @@
+from .base import BaseAuth
+from ..exceptions import AuthException, AuthTokenError
+import hmac
+from base64 import b64encode, b64decode
+from hashlib import sha256
+import urllib
+from ..utils import parse_qs
+import time
+
+
+class DiscourseAuth(BaseAuth):
+    name = "discourse"
+    EXTRA_DATA = ["username", "name", "avatar_url"]
+
+    def auth_url(self):
+        """Return redirect url"""
+        returnUrl = self.redirect_uri
+        nonce = self.strategy.random_string(64)
+        self.add_nonce(nonce)
+
+        payload = "nonce=" + nonce + "&return_sso_url=" + returnUrl
+        base64Payload = b64encode(payload)
+        payloadSignature = hmac.new(
+            self.setting("SOCIAL_AUTH_DISCOURSE_AUTH_SECRET"), base64Payload, sha256
+        ).hexdigest()
+        encodedParams = urllib.urlencode(
+            {"sso": base64Payload, "sig": payloadSignature}
+        )
+        return (
+            self.setting("SOCIAL_AUTH_DISCOURSE_AUTH_SERVER_URL")
+            + "/session/sso_provider?"
+            + encodedParams
+        )
+
+    def get_user_id(self, details, response):
+        return response["email"]
+
+    def get_user_details(self, response):
+        results = {
+            "username": response.get("username"),
+            "email": response.get("email"),
+            "name": response.get("name"),
+            "groups": response.get("groups", "").split(","),
+            "is_staff": response.get("admin") == "true"
+            or response.get("moderator") == "true",
```

Comment:
Could you:

1. Move the `or` to the previous line
2. Indent the `response.get...` to the same level as the previous one

?